contractcourt/chain_arbitrator.go

```diff
@@ -155,6 +159,11 @@ type ChainArbitrator struct {
 	// that are still considered open.
 	activeWatchers map[wire.OutPoint]*chainWatcher
 
+	// newChainWatcher is used by the closedChanNotifier goroutine to find
+	// out about new chain watchers and subscribe to their channel event
+	// notifications, which will be passed on to the chanNotifier.
+	newChainWatcher chan *chainWatcher
```

Comment:
I think a better approach is to utilize the `MarkChannelClosed` closure to get the close events.

Reply:
agree, implemented!